.travis.yml

```diff
@@ -0,0 +1,71 @@
+sudo: required
+services:
+  - docker
+
+language: python
+
+cache:
+    directories:
+      - $HOME/.cache
+
+git:
+    depth: 100500
+
+env:
+    global:
+      - PRODUCT=tarantool-luacheck
+    matrix:
+      - OS=el DIST=6
+      - OS=el DIST=7
+      - OS=fedora DIST=26
+      - OS=fedora DIST=27
+
+script:
+  - git describe --long
+  - git clone https://github.com/packpack/packpack.git packpack;
+  - VERSION=`git describe --tags | sed 's/\(.*\)-.*/\1/' | sed 's/[^0-9-]*//g' | \
+    cut -c 2- | sed 's#-#\.#g'`
```

Comment:
Are not that simpler?

```
git describe --tags | sed 's/^[a-z]\+-\([0-9]\+\)-\([0-9]\+\)-\([0-9]\+\)$/\1.\2/'
```

Reply:
Doesn't seem to do the job.
```
git describe --tags | sed 's/\(.*\)-.*/\1/' | sed 's/[^0-9-]*//g' | cut -c 2- | sed 's#-#\.#g'
2.9.0.7
git describe --tags | sed 's/^[a-z]\+-\([0-9]\+\)-\([0-9]\+\)-\([0-9]\+\)$/\1.\2/'
rel-2-9-0-7-gfb9a828
```

Reply:
Ok, I missed case when you are not on the tag itself. Maybe this will be simpler and fit your needs:

```
git describe --all --long | sed -e 's/^.*rel-//' -e 's/-g[a-f0-9]\+$//' -e 's/-/./g'
2.9.0.0
```

Reply:
Hm.
```
git checkout master
git describe --all --long | sed -e 's/^.*rel-//' -e 's/-g[a-f0-9]\+$//' -e 's/-/./g'
2.9.0.0.g6511dc6
```
And
```
git checkout rpm-build-deploy
git describe --all --long | sed -e 's/^.*rel-//' -e 's/-g[a-f0-9]\+$//' -e 's/-/./g'
heads/rpm.build.deploy.0.g11bcf4c
```

Reply:
rpm-build-deploy is the branch? It seems that it is considered as a tag with --all? Anyway, I tried to simplify, feel free to use any approach.